libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -29,8 +31,65 @@ public enum Mode {
         READ_WRITE
     }
 
-    public record FileData(String path, Mode mode) {
+    public enum BaseDir {
+        CONFIG,
+        DATA
+    }
+
+    public static final class FileData {
```

Comment:
Any reason this is not a record anymore?

Reply:
Since the combination of acceptable arguments is fixed, I wanted to provide methods to build it correctly (avoid possible "wrong" combinations by design), and make the constructor private.

Reply:
That might be better done with checks or assertions inside the canonical record constructor. You can still have factory methods for convenience, of course.

Like:

```
public FileData {
    if (path == null) {
        assert relativePath != null && baseDir != null;
    } else {
        assert relativePath == null && baseDir == null;
    }
}
```

We could also consider a model with algebraic types, rather than implicitly inferring "absoluteness" from the field values:

```
sealed interface FileData {
    Mode mode();

    record AbsoluteFileData(Mode mode, Path path) implements FileData {};
    record RelativeFileData(Mode mode, BaseDir baseDir, Path relativePath) implements FileData {};
}
```

Reply:
(Still not a record as I wanted the 2 implementations to have each their own private fields and expose only mode + resolved path)